pkg/config/parse.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"io/ioutil"
+	"runtime"
+
+	"github.com/BurntSushi/toml"
+	"github.com/kelseyhightower/envconfig"
+	validator "gopkg.in/go-playground/validator.v9"
+)
+
+var validate *validator.Validate
+
+func getValidator() *validator.Validate {
+	if validate == nil {
+		return validator.New()
+	}
+	return validate
+}
+
+// Config provides configuration values for all components
+type Config struct {
+	GoEnv                string         `validate:"required" envconfig:"GO_ENV" default:"development"`
+	GoBinary             string         `validate:"required" envconfig:"GO_BINARY_PATH" default:"go"`
+	LogLevel             string         `validate:"required" split_words:"true" default:"debug"`
+	MaxConcurrency       int            `validate:"required" split_words:"true"`
+	MaxWorkerFails       uint           `validate:"required" split_words:"true" default:"5"`
+	CloudRuntime         string         `validate:"required" split_words:"true" default:"none"`
+	FilterFile           string         `validate:"required" split_words:"true" default:"filter.conf"`
+	Timeout              int            `validate:"required" default:"300"`
+	EnableCSRFProtection bool           `envconfig:"ATHENS_ENABLE_CSRF_PROTECTION" default:"false"`
+	Proxy                *ProxyConfig   `validate:""`
+	Olympus              *OlympusConfig `validate:""`
+	Storage              *StorageConfig `validate:""`
+}
+
+// ParseConfigFile parses the given file into an athens config struct
+func ParseConfigFile(configFile string) (*Config, error) {
+
+	// attempt to read the given config file
+	confBytes, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		return nil, err
+	}
+	confStr := string(confBytes)
+	return parseConfig(confStr)
+}
+
+func parseConfig(confStr string) (*Config, error) {
+	var config Config
+	if _, err := toml.Decode(confStr, &config); err != nil {
```

Comment:
you can get rid of all the stuff in lines 41-46 and just use `toml.DecodeFile` and pass the file path :)

Reply:
I broke it apart to make it a little easier to test, but I'm happy to use `DecodeFile` too if you think that's better :)